Extended AJAX JSON support and optional vaolue encode

itbeyond opened the pull request:
Added the contentType ajax option defaults to 'application/x-www-form-urlencoded' can be changed to 'application/json; charset=utf-8' which is required for .net web service support. To support this
this the response must check data.d not just data - this has been added so both responses are supported..

Also to support the use of a URL (or other non encode required values) as the returned value the
encodeURIComponent in the buildItems function must be able disabled, so the new option encodeValue was added this defaults to true which means the value is encoded - set to false to support URL or non encode required values.

itbeyond committed:
Added options contentType, encodeValue, full POST json support and support URL as value

,

Added the contentType ajax option defaults to
'application/x-www-form-urlencoded' can be changed to 'application/json;
charset=utf-8' which is required for .net web service support. To make
this work the response must be changed for data.d - this has been added.

Also to support the use of a URL as the returned value the
encodeURIComponent in the buildItems function must be able disabled, so
the option encodeValue was added this defaults to true which means the
value is encoded - set to false to support URL or non encode required
values.

commadelimited commented:
@itbeyond...some of the changes you've made sound very intriguing. But the way you submitted your pull request means that I can't compare on a line by line basis. Would you mind tweaking your change and resubmitting?

itbeyond committed:
Revert "Added options contentType, encodeValue, full POST json support and support URL as value"

This reverts commit a938f1767cfa9f8a83b8317fa7e16b14cfb3bf57.

itbeyond committed:
Andy, This should be a line by line commit for the pull
Added options contentType, encodeValue, full POST json support and support URL as value

,

Added the contentType ajax option defaults to
'application/x-www-form-urlencoded' can be changed to 'application/json;
charset=utf-8' which is required for .net web service support. To make
this work the response must be changed for data.d - this has been added.

Also to support the use of a URL as the returned value the
encodeURIComponent in the buildItems function must be able disabled, so
the option encodeValue was added this defaults to true which means the
value is encoded - set to false to support URL or non encode required
values

commadelimited commented:
Hey @itbeyond, would you mind rebasing and resubmitting your pull request. Some other releases have been made since this was put in which make it so that I can't auto-merge your changes.
